### PR TITLE
Replace bug report and feature request template with a GitHub Forms YAML template [#1874]

### PR DESCRIPTION
**Pull request by kruncher**
### What is the context of this PR?
This PR replaces the "Bug Report" markdown template with a [GitHub Forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms) YAML template.

### How to review
- Take a look at the new bug report template: https://github.com/ONSdigital/design-system/blob/38eaafe123ea84ee4c19899ec6f6758d8e0f3aff/.github/ISSUE_TEMPLATE/bug_report.yaml
- Take a look at the new feature request template: https://github.com/ONSdigital/design-system/blob/da01958296e95a5db37b2fb79a6c80624b470a4d/.github/ISSUE_TEMPLATE/feature_request.yaml

### Related issues
- #1874

### CODE REVIEW

**Review by boxadesign — APPROVED** [review]

**Comment by kruncher** [comment]
Added new feature request template as well.

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      description: Add screenshot(s) or examples of the bug
+      placeholder: Add a file, image or link
+    validations:
+      required: false
+  - type: input
+    id: version
+    attributes:
+      label: Version
+      description: Confirmation of the version of the design system that is being used
+      placeholder: ex. 32.2.7
```
Comment (jrbarnes9) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:52]:
```suggestion
      placeholder: For example, 32.2.7
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    id: device-details
+    attributes:
+      label: Device Details
+      description: Here you can provide further information about the device(s) that are affected by this bug/issue.
+    validations:
+      required: false
+  - type: checkboxes
+    id: operating-system
+    attributes:
+      label: Operating System
```
Comment (jrbarnes9) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:74]:
```suggestion
      label: Operating system
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        - label: Firefox
+        - label: Chrome
+        - label: Safari
+        - label: Microsoft Edge
+        - label: Microsoft Internet Explorer
+        - label: Other
+  - type: input
+    id: browsers-details
+    attributes:
+      label: Browser Details
```
Comment (jrbarnes9) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:103]:
```suggestion
      label: Browser details
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+      options:
+        - label: Windows
+        - label: macOS
+        - label: Android
+        - label: iOS
+        - label: Other
+  - type: input
+    id: operating-system-details
+    attributes:
+      label: Operating System Details
```
Comment (jrbarnes9) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:85]:
```suggestion
      label: Operating system details
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 5]
```diff
@@ -0,0 +1,106 @@
+name: Bug Report
+description: File a bug report
+title: "[Bug]: "
+labels: ["bug", "triage"]
+assignees: []
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to fill out this bug report!
```
Comment (jrbarnes9) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:10]:
```suggestion
        Thank you for taking the time to fill out this bug report
```

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+description: Propose a new component, pattern, style or documentation to add to the ONS Design System
+title: "[Feature]: "
+labels: ["New feature proposal", "triage"]
+assignees: []
+body:
+  - type: markdown
+    attributes:
+      value: |
+        **Before you fill in this proposal template:**
+        - If you are suggesting a change to a component, pattern or style that is already in the ONS Design System, please comment on its GitHub discussion instead (https://github.com/ONSdigital/design-system/discussions). You can get to the discussion using the direct link at the bottom of each page in the Design System.
```
Comment (jrbarnes9) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:11]: Will the template let us hyperlink and hide the url?
```suggestion
        - If you are suggesting a change to a component, pattern or foundation that is already in the ONS Design System, please comment on its [GitHub discussion instead] (https://github.com/ONSdigital/design-system/discussions). You can get to the discussion using the direct link at the bottom of each page in the Design System.
```

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        - If you are proposing something new, check the ONS Design System community backlog first (https://github.com/ONSdigital/design-system/projects/1). There may already be a proposal for something similar. You can add examples of your own evidence to an existing proposal.
```
Comment (jrbarnes9) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:12]:
```suggestion
        - If you are proposing something new, check the [ONS Design System community backlog](https://github.com/ONSdigital/design-system/projects/1) first. There may already be a proposal for something similar. You can add examples of your own evidence to an existing proposal.
```

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 8]
```diff
@@ -0,0 +1,45 @@
+name: New feature proposal
+description: Propose a new component, pattern, style or documentation to add to the ONS Design System
```
Comment (jrbarnes9) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:2]:
```suggestion
description: Propose a new component, pattern, foundation or documentation to add to the ONS Design System
```

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  - type: markdown
+    attributes:
+      value: |
+        **Before you fill in this proposal template:**
+        - If you are suggesting a change to a component, pattern or style that is already in the ONS Design System, please comment on its GitHub discussion instead (https://github.com/ONSdigital/design-system/discussions). You can get to the discussion using the direct link at the bottom of each page in the Design System.
+        - If you are proposing something new, check the ONS Design System community backlog first (https://github.com/ONSdigital/design-system/projects/1). There may already be a proposal for something similar. You can add examples of your own evidence to an existing proposal.
+  - type: textarea
+    id: what
+    attributes:
+      label: What
```
Comment (jrbarnes9) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:16]:
```suggestion
      label: What feature would you like to add to the ONS Design System?
```

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+      description: Give a brief description of the new feature you want to propose for the Design System. This could be a component, pattern or foundation, or new documentation for an existing feature.
+    validations:
+      required: true
+  - type: textarea
+    id: why
+    attributes:
+      label: Why
```
Comment (jrbarnes9) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:23]:
```suggestion
      label: Why should this new feature be added to the Design System?
```

**Review by jrbarnes9 — COMMENTED** [review]
Think I'd be inclined to keep the labels consistently as questions e.g.
"What devices are you seeing the problem on?"
"What version of the Design System are you using?"
"What operating systems are you seeing the problem on?"
"How is this problem impacting your service?"

Don't think you'd always need the "Details of"/"Confirmation of..." hint text then.

Could be more consistent with "bug or issue", rather than "bug/issue"

**Comment by kruncher** [comment]
@jrbarnes9 I have applied your suggested changes.

**Review by boxadesign — APPROVED** [review]

**Comment by jrbarnes9** [comment]
> Think I'd be inclined to keep the labels consistently as questions e.g. "What devices are you seeing the problem on?" "What version of the ONS Design System are you using?" "What operating systems are you seeing the problem on?" "How is this problem impacting your service?"
> 
> Don't think you'd always need the "Details of"/"Confirmation of..." hint text then.
> 
> Could be more consistent with "bug or issue", rather than "bug/issue"

@kruncher Could you do these changes too?

**Review by jrbarnes9 — APPROVED** [review]

**Review by boxadesign — APPROVED** [review]

**Pull request merged**